android/app/src/main/java/com/cBreez/client/BreezSdkWorker.kt

```diff
@@ -0,0 +1,160 @@
+        if (e is BreezEvent.InvoicePaid) {
+            Log.v(
+                TAG,
+                "Received payment. Bolt11: ${e.details.bolt11}\nPayment Hash:${e.details.paymentHash}"
+            )
+            e.details.payment?.let {
+                paymentListener(it)
+            }
+        }
+    }
+}
+
+class BreezSdkWorker(appContext: Context, workerParams: WorkerParameters) :
+    CoroutineWorker(appContext, workerParams) {
+
+    override suspend fun getForegroundInfo(): ForegroundInfo {
+        return ForegroundInfo(
+            NOTIFICATION_ID_PAYMENT_RECEIVED,
+            createNotification(
+                "Incoming payment",
+                NotificationManager.IMPORTANCE_LOW,
+                true
+            )
+        )
+    }
+
+    override suspend fun doWork(): Result = coroutineScope {
+        // TODO(_): Improve error handling
+        val breezSDK = connectSDK(::paymentListener)
+        val paymentHash =
+            inputData.getString("PAYMENT_HASH") ?: return@coroutineScope Result.failure()
+        val payment = breezSDK?.paymentByHash(paymentHash)
+        if (payment != null && payment.status == PaymentStatus.COMPLETE) {
+            createNotification("Received ${payment.amountMsat / 1000u} sats")
+            return@coroutineScope Result.success()
+        } else {
+            createNotification("Receive payment failed")
+            return@coroutineScope Result.failure()
+        }
+    }
+
+    private fun paymentListener(payment: Payment): Void {
+        // TODO(_): Improve error handling
+        payment.details.let {
+            if (it is PaymentDetails.Ln) {
+                val paymentHash =
+                    inputData.getString("PAYMENT_HASH")
+                if (paymentHash != null) {
+                    if (it.data.paymentHash == paymentHash) {
+                        createNotification("Received ${payment.amountMsat / 1000u}) sats")
+                    }
+                }
+            }
+        }
+        return TODO("Provide the return value")
+    }
+
+    private fun connectSDK(paymentListener: (payment: Payment) -> Void): BlockingBreezServices? {
+        // TODO(_): Improve error handling
+        // Select your seed, invite code and environment
+        // TODO(_): Read mnemonic from Keystore
+        val mnemonic: String = "<your-mnemonic>"
+        val seed = mnemonicToSeed(mnemonic)
+        val apiKey = applicationContext.getString(R.string.breezApiKey)
+
+        // Create the default config
+        val glNodeConf = GreenlightNodeConfig(null, null)
+        val nodeConf = NodeConfig.Greenlight(glNodeConf)
+        val config = defaultConfig(EnvironmentType.PRODUCTION, apiKey, nodeConf)
+        // TODO(_): Set workingDir programmatically
+        config.workingDir = "/data/user/0/com.cBreez.client.debug/app_flutter"
+        // Connect to the Breez SDK make it ready for use
+        return connect(config, seed, SDKListener(paymentListener))
```

Comment:
Perhaps it is better to use a static instance of the BlockingBreezServices? like we do in iOS?

Reply:
Done as part of [BreezSdkWorker second draft](https://github.com/breez/c-breez/pull/737/commits/253c8520b683d788c875ba66709a8b55f592fc57)